modules/repository/license_test.go

```diff
@@ -179,3 +181,62 @@ Copyright (C) 2023 by Gitea teabot@gitea.io
 		})
 	}
 }
+
+func Test_detectLicense(t *testing.T) {
+	type DetectLicenseTest struct {
+		name string
+		arg  string
+		want []string
+	}
+
+	tests := []DetectLicenseTest{
+		{
+			name: "empty",
+			arg:  "",
+			want: nil,
+		},
+		{
+			name: "no detected license",
+			arg:  "Copyright (c) 2023 Gitea",
+			want: nil,
+		},
+	}
+
+	LoadRepoConfig()
+	err := loadLicenseAliases()
+	assert.NoError(t, err)
+	for _, licenseName := range Licenses {
+		license, err := GetLicense(licenseName, &LicenseValues{
+			Owner: "Gitea",
+			Email: "teabot@gitea.io",
+			Repo:  "gitea",
+			Year:  time.Now().Format("2006"),
```

Comment:
time.Now() in tests are dangerous ...
TODO: need to look why it's used

Reply:
looks ok in my view, this just using `GetLicense` to generate test input data. different years willn't affect test results.

Reply:
Fixed.